plasmapy/diagnostics/thomson.py

```diff
@@ -431,12 +445,24 @@ def spectral_density(
         if np.sum(ifract) != 1:
             raise ValueError(f"The provided ifract does not sum to 1: {ifract}")
 
+    # if both scattering_angle and probe_vec or scatter_vec, raise error
+    if scattering_angle is not None and (
+        scatter_vec is not None or probe_vec is not None
+    ):
+        raise ValueError(
+            "Keywords scattering_angle and scatter_vec or probe_vec are both given."
+            "Give only scattering_angle or scatter_vec and probe_vec"
+        )
+
     if probe_vec is None:
```

Comment:
The more I think about it, having default values for these might be confusing. Now that we have the nice scattering angle keyword, I'd suggest removing these two default value blocks and just add a third block that ensures that at least scattering_angle OR probe_vec and scatter_vec are provided.